packages/synapse-react-client/src/components/SynapseTable/usePrefetchTableData.ts

```diff
@@ -2,45 +2,51 @@ import {
   ColumnTypeEnum,
   FileHandleAssociateType,
   FileHandleAssociation,
+  QueryBundleRequest,
   ReferenceList,
   RestrictableObjectType,
+  RowSet,
+  Table,
 } from '@sage-bionetworks/synapse-types'
 import { isEntityViewOrDataset, isFileViewOrDataset } from './SynapseTableUtils'
 import {
+  useGetEntity,
   useGetEntityHeaders,
   useGetFileBatch,
   useGetRestrictionInformationBatch,
   useGetUserGroupHeaders,
 } from '../../synapse-queries'
-import { useAtomValue } from 'jotai'
-import {
-  tableQueryDataAtom,
-  tableQueryEntityAtom,
-} from '../QueryWrapper/QueryWrapper'
 import { getFieldIndex, getTypeIndices } from '../../utils/functions/queryUtils'
-import { useMemo } from 'react'
+import { useQueryContext } from '../QueryContext'
+import { useQuery, useQueryClient } from '@tanstack/react-query'
+import { useCallback, useMemo } from 'react'
+import { useSynapseContext } from '../../utils'
+import { goToPage as transformQueryToGoToPage } from '../../utils/hooks/useImmutableTableQuery/TableQueryReducerActions'
+import { cloneDeep } from 'lodash-es'
+import { getTableQueryUseQueryOptions } from '../QueryWrapper/TableQueryUseQueryOptions'
+
+function usePrefetchFileHandleData(rowSet: RowSet) {
```

Comment:
Updated to pass RowSet as an argument, and use queryContext + react-query instead of Jotai. No logical changes